collectors/mesos/agent/agent.go

```diff
@@ -102,9 +102,14 @@ func (c *Collector) metricsMessages() (out []producers.MetricsMessage) {
 	t := time.Unix(c.timestamp, 0)
 
 	for _, cm := range c.containerMetrics {
+		datapoints, err := c.createContainerDatapoints(cm)
+		if err != nil {
+			c.log.Warnf("Could not create datapoints for container ID %s: %s", cm.ContainerID, err)
```

Comment:
Note, you include the container ID in the error message that could come from this call, so may be redundant to include it here as well.